### PR TITLE
Polygon ecosystem DKG coordination (Alpha 4)

### PR DESCRIPTION
**Pull request by KPrasch**
compatibility changes for polygon ecosystem deployments of dkg coordination, although to bulk of the code supports event scanning.

This PR needs additional follow-up to make the tests pass, but it's okay to review now!

Co-Authored by: 
@derekpierre 
@cygnusv 
@KPrasch

### CODE REVIEW

**Review by derekpierre — APPROVED** [review]
🎸

File: nucypher/blockchain/eth/trackers/dkg.py [hunk 1]
```diff
@@ -253,7 +261,10 @@ def scan(self):
         # Note that our chain reorg safety blocks cannot go negative
         if self.start_block is None:
             self.start_block = self._get_start_block_number()
-        start_block = max(self.state.get_last_scanned_block() - chain_reorg_safety_blocks, self.start_block)
+        start_block = min(
+            self.state.get_last_scanned_block() - chain_reorg_safety_blocks,
```
Comment (derekpierre) [inline nucypher/blockchain/eth/trackers/dkg.py:265]: Hmmm 🤔 ...what's the value of `get_last_scanned_block()` on startup?

`start_block` is only set once on startup, so using `min` here could cause it to stay at the same block forever, as the scan tries to progress over time. That's why `max` was used.

**Review by derekpierre — CHANGES_REQUESTED** [review]
Just drawing attention to my comment about `start_block`

**Review by derekpierre — DISMISSED** [review]
Just drawing attention to my comment about `start_block`

File: nucypher/blockchain/eth/trackers/dkg.py (hunk 1 above)
Comment (cygnusv) [reply nucypher/blockchain/eth/trackers/dkg.py:265]: Ok, that makes sense, but something's weird here. The behavior that prompted us to look here was that in the logs, we saw patterns were blocks seemed to be skipped like the following:
```
Scanning blocks: 1234 - 1234
... 
Scanning blocks: 1244 - 1244
...
Scanning blocks: 1255 - 1254   (yeah sometimes the start block was higher than the end block)
```
Also, the code in this function seem overly complex and trying to do many things at the same time.

Comment (derekpierre) [reply nucypher/blockchain/eth/trackers/dkg.py:265]: We also need some improved testing in this area.

**Comment by KPrasch** [comment]
rebased @ 5c4cc708c7f352d7a5c7751a4ad7ff6a9f1b8082

File: tests/integration/learning/test_domains.py [hunk 2]
```diff
@@ -124,6 +131,7 @@ def test_learner_ignores_stored_nodes_from_other_domains(
     assert pest not in other_staker.known_nodes  # But not anymore.
 
 
+@pytest.mark.skip
```
Comment (piotr-roslaniec) [inline tests/integration/learning/test_domains.py:134]: What is the reason for skipping these tests? I assume it's temporary?

Comment (KPrasch) [reply tests/integration/learning/test_domains.py:134]: Ah yes, thanks for the reminder, I'll attempt to restore these tests before merging.

File: nucypher/blockchain/eth/clients.py [hunk 3]
```diff
@@ -63,11 +63,13 @@ class Web3ClientUnexpectedVersionString(Web3ClientError):
 
 # TODO: This list is incomplete, but it suffices for the moment - See #1857
 POA_CHAINS = {
-    4,    # Rinkeby
-    5,    # Goerli
-    42,   # Kovan
-    77,   # Sokol
+    4,  # Rinkeby
```
Comment (cygnusv) [inline nucypher/blockchain/eth/clients.py:66]: Rinkeby is officially dead ☠️

**Review by cygnusv — APPROVED** [review]

**Pull request merged**